python/ray/autoscaler/autoscaler.py

```diff
@@ -462,18 +465,28 @@ def launch_new_node(self, count):
                 TAG_RAY_NODE_STATUS: "uninitialized",
                 TAG_RAY_LAUNCH_CONFIG: self.launch_hash,
             }, count)
+        with self.launch_lock:
+            self.num_launches_pending -= count
```

Comment:
Shall we do this in a finally block for the create_node call?

Reply:
Good point. This function no longer exists but I've used a try-finally block in the new NodeLauncher class.